### DISCUSSION TITLE
Elsa 3.0.5 - Get Json Data from body HTTP end point and write and read it in variable

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5542] [Q&A]
hello, sorry for my double question, i'm still facing ame issued in case get data from json body in http end point then write in varible then read it and return back as http response, i'm confuse how to read and write data in variable,

i found it's kinda different betwen documentation and sample json that i exported example in variable :
`{
            "variable":{
               "id":"d2920be3dbbf4826b7d2dc4d41f5f509",
               "name":"NewUser",
               "typeName":"Object"
            },
            "value":{
               "typeName":"Object",
               "expression":{
                  "type":"Literal"
               }
            }
}` 
i dont understant in part poroperty value due in documetation the struture value property did'n same,
so can i have some example with elsa 3 any vaersion how to read and write variable from http end point

**Comment by glime-ai** [comment]
To read and write variables from an HTTP endpoint in Elsa 3, you can follow these steps:

1. **Define the Variable**: First, you need to define the variable in your workflow. This can be done programmatically or using the visual designer. In your JSON, it looks like you have defined a variable named `NewUser` with a type of `Object`.

2. **Read JSON Body**: To read data from the JSON body of an HTTP request, you can use the `HttpEndpoint` activity with the `ReadContent` property set to `true`. This will parse the incoming JSON body and make it available for use in the workflow.

3. **Assign to Variable**: To assign the data from the JSON body to a variable, you can use the `SetVariable` activity. You will specify the name of the variable and use an expression to assign the value. For example, if the JSON body contains a user object, you might use an expression like `input.Body` to assign the entire JSON body to the variable.

4. **Use the Variable**: Once the variable is set, you can use it in subsequent activities. For example, you can pass it to a `WriteHttpResponse` activity to return it as an HTTP response.

Here's a simplified example of what the JSON configuration might look like for these steps:

```json
{
  "activities": [
    {
      "type": "HttpEndpoint",
      "properties": {
        "Path": "/my-endpoint",
        "Methods": ["POST"],
        "ReadContent": true
      }
    },
    {
      "type": "SetVariable",
      "properties": {
        "VariableName": "NewUser",
        "ValueExpression": "input.Body"
      }
    },
    {
      "type": "WriteHttpResponse",
      "properties": {
        "Content": "{{ Variables.NewUser }}",
        "ContentType": "application/json",
        "StatusCode": 200
      }
    }
  ]
}
```

In this example, the `HttpEndpoint` activity reads the JSON body from a POST request, the `SetVariable` activity assigns the JSON body to the `NewUser` variable, and the `WriteHttpResponse` activity returns the variable as a JSON response.

Please note that the actual JSON configuration might differ based on the specifics of your workflow and the version of Elsa you are using. For detailed examples and documentation, you can refer to the official Elsa documentation and the sample workflows provided.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
